index.js

```diff
@@ -0,0 +1,258 @@
+// @flow
+
+import GeocoderArcGIS from 'geocoder-arcgis'
+import lonlat from '@conveyal/lonlat'
+
+const arcGisGeocoders = {}
+
+type Boundary = {
+  rect: {
+    minLat: number,
+    minLon: number,
+    maxLat: number,
+    maxLon: number
+  }
+}
+
+type BaseQuery = {
+  clientId?: string,
+  clientSecret?: string,
+  url?: string
+}
+
+type BaseResponse = {
+  query: any
+}
+
+function boundaryToSearchExtent (boundary: Boundary): string {
+  return [
+    boundary.rect.minLon,
+    boundary.rect.maxLat,
+    boundary.rect.maxLon,
+    boundary.rect.minLat
+  ].join(',')
+}
+
+/**
+ * To make flow happy
+ */
+function coerceNullToString (s: ?string) {
+  return s || ''
+}
+
+function getGeocoder (clientId: ?string, clientSecret: ?string, endpoint: ?string) {
```

Comment:
Have you run into a use case where you have more than one ArcGIS geocoder?

Reply:
Not really, but with the way I setup react-select-geocoder-arcgis the keys are sent as props, so theoretically they could change.